src/bin/sol-fbp-generator/main.c

```diff
@@ -233,8 +232,7 @@ read_meta_type_file(const struct sol_flow_metatype_context *ctx,
     }
 
     contents = sol_file_reader_get_all(reader);
-    *buf = contents.data;
-    *size = contents.len;
+    *buf = SOL_BUFFER_INIT_CONST((char *)contents.data, contents.len);
```

Comment:
append the data will make you duplicate the data. I'm ok with this since you have sure about the life-cycle of this memory, i mean, the buffer will not be used after the sol_file_reader be released.

@kaspersky this is not c++ the cast is not necessary i think.